### DISCUSSION TITLE
Maturin as a backend in select projects for mixed Rust Python projects

### DISCUSSION THREAD

**Discussion by nazq** [DavidVujic/python-polylith#297] [Ideas]
I'm looking to migrate a large mixed Rust/Python code base (all using uv) to a monorepo structure, and wanted the polylith structure if possible over and above uv workspaces. One thing I'd need to make this happen is to support [maturin](https://github.com/PyO3/maturin) as a build backend. Has any thought been given to this ?

**Comment by DavidVujic** [comment]
Hi @nazq! I don't know much about Maturin (yet) and will check it out. If it has similar capabilities as `hatchling` (such as build hooks or plugin support) then I think it is possible to support it for sure!

I will read up on Maturin and see what can be done 😄

**Comment by DavidVujic** [comment]
[This part](https://www.maturin.rs/config#maturin-options) looks promising (the `include`), I will test that out.

**Comment by DavidVujic** [comment]
@nazq  A short update: I have started working on a solution for Maturin and other Package & Dependency management tools without support for plugins or build hooks. I _think_ I will be able to have something ready within a couple of days. 😄

**Comment by DavidVujic** [comment]
@nazq Announcing Maturin support here: https://github.com/DavidVujic/python-polylith/discussions/311

**Reply by nazq** [reply]
Great to see. I'll take it for a spin this week and see how it drives. Many thanks

**Reply by DavidVujic** [reply]
That would be great! I am not a Maturin user and would appreciate help with testing this workflow out 🙏

**Comment by gregunz** [comment]
@nazq: would be curious to know what you ended up doing, have been considering to include some rust in my monorepo as well and was wondering if that even makes sense!
Any findings would be useful :)

**Reply by nazq** [reply]
I'm all monorepo now. Both rust and python, even when they share no code directly but one service relies on the other. Tbh i flipped from being 90/10 py/rust to being 5/95. Cargo does a way better job with mono repos so i barely think about it anymore

**Reply by gregunz** [reply]
so you are really moslty rust + cargo, and python is rather hanging around in some places? no polylith bases or components I understand?

Would you still recommend it some a project that is 95/5 (the opposite of you)? or any suggestions into having it in this monorepo structure?
